core/ustring.cpp

```diff
@@ -1325,8 +1325,10 @@ String String::num_real(double p_num) {
 			dec_int /= 10;
 		}
 		sd = '.' + decimal;
-	} else {
+	} else if (p_trailing) {
```

Comment:
Do we really want to make this optional though? IMO it would be just as well to enforce it, so that it's always clear that it's a real and not an integer.

Reply:
Either is fine to me. I made it this way because it looks cleaner, but maybe it's better to always show it to make it clear that it's a real.

P.S. In C#, the default behavior of converting numbers to string does not include the trailing decimals, so if we want them I'd add a format specifier (should be something like `"0.0"`).